### PR TITLE
International addresses show the wrong error messages

### PR DESCRIPTION
**Pull request by JonF1**
## Context

If validations are triggered on submission of course details form, the error messages for both international and uk form fields is identical. This is incorrect because the field labels are different despite the attributes being the same.
## Changes proposed in this pull request
* For the international address I've added custom validations to allow different error messaging to be rendered as shown below for both blank and length. 

Before: 

Screenshot 2021-11-23 at 18 07 12

After:
Screenshot 2021-11-23 at 18 08 20

## Guidance to review
Trigger validations on `/candidate/application/contact-information/address/edit` (or new) endpoint

## Link to Trello card
https://trello.com/c/iqMeSjy0/3571-international-addresses-show-the-wrong-error-messages-if-left-empty-next-sprint-check)

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JonF1** [comment]
[International addresses show the wrong error messages if left empty (Next sprint & Check )](https://trello.com/c/iqMeSjy0/3571-international-addresses-show-the-wrong-error-messages-if-left-empty-next-sprint-check)

File: app/forms/candidate_interface/contact_details_form.rb [hunk 1]
```diff
@@ -2,16 +2,21 @@ module CandidateInterface
...
+
+    validate :address_line1_blank?, on: :address, if: :international?
+    validate :address_lines_length_valid?, if: :international?
+
     validates :address_type, presence: true, on: :address_type
     validates :country, presence: true, on: :address_type, if: :international?
 
     validates :address_line1, :address_line2, :address_line3, :address_line4,
-              length: { maximum: 50 }, on: :address
+              length: { maximum: 50 }, on: :address, if: :uk?
```
Comment (stevehook) [inline app/forms/candidate_interface/contact_details_form.rb:19]: Can you use the `MAX_LENGTH` constant you added here?

File: app/forms/candidate_interface/contact_details_form.rb [hunk 2]
```diff
@@ -2,16 +2,21 @@ module CandidateInterface
   class ContactDetailsForm
     include ActiveModel::Model
 
+    MAX_LENGTH = 50
```
Comment (stevehook) [inline app/forms/candidate_interface/contact_details_form.rb:5]: :+1:

File: config/locales/candidate_interface/contact_details.yml [hunk 3]
```diff
@@ -50,14 +50,19 @@ en:
               too_long: Phone number must be %{count} characters or fewer
             address_line1:
               blank: Enter your building and street
+              international_blank: Enter your address
               too_long: Building and street line 1 must be %{count} characters or fewer
+              international_too_long: Address line 1 must be 50 characters or fewer
```
Comment (stevehook) [inline config/locales/candidate_interface/contact_details.yml:55]: Could you use `%{count}` in these new strings to avoid the hard-coded `50`s

**Review by stevehook — COMMENTED** [review]
Looks good so far. A couple of minor suggestions inline.

File: config/locales/candidate_interface/contact_details.yml (hunk 3 above)
Comment (JonF1) [reply config/locales/candidate_interface/contact_details.yml]: Yep, I think `%{count}` is given to us by max option. So I've added count argument to get the same behaviour 👍

**Review by stevehook — APPROVED** [review]

**Pull request merged**